Samples/WebDemo/Controllers/EvaluationController.cs

```diff
@@ -42,7 +42,13 @@ public IActionResult HandleRequest([FromBody] RequestBody body)
                 var parameters = (RecordValue) FormulaValue.FromJson(body.context);
                 
                 // TODO - this should be eval Async and catch timeouts. 
-                var result = engine.Eval(body.expression, parameters);
+                if (parameters == null)
+                {
+                    parameters = RecordValue.Empty();
+                }
+                var check = engine.Check(body.expression, parameters.Type, options:null);
+                check.ThrowOnErrors();
+                var result =  check.Expression.EvalAsync(parameters, cancel: CancellationToken.None).Result;
```

Comment:
>.Result [](http://example.com/codeflow?start=100&length=7)

We could actually get rid of this, and make the method be 

async Task<IActionResult>

and then await this. 

Then we could also add cancellation support, have it just abort after 5 seconds. 
See test case here:
[https://github.com/microsoft/Power-Fx/blob/25e9106435fc6c4fa2e8208bea192d8a02dfcead/src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineAsyncTests.cs#L173](https://github.com/microsoft/Power-Fx/blob/25e9106435fc6c4fa2e8208bea192d8a02dfcead/src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineAsyncTests.cs#L173)

Reply:
As Discussed, added abort after 2 seconds.